tools/myst/gcov/myst.edl

```diff
@@ -0,0 +1,513 @@
+            int flags);
+
+        long myst_sendmsg_ocall(
+            int sockfd,
+            /* -- begin struct msghdr -- */
+            [in, size=msg_namelen] const void* msg_name,
+            socklen_t msg_namelen,
+            [in, size=len] const void* buf,
+            size_t len,
+            [in, size=msg_controllen] const void* msg_control,
+            socklen_t msg_controllen,
+            int msg_flags,
+            /* -- end struct msghdr -- */
+            int flags)
+            transition_using_threads;
+
+        long myst_recvmsg_ocall(
+            int sockfd,
+            /* -- begin struct msghdr -- */
+            [out, size=msg_namelen] void* msg_name,
+            socklen_t msg_namelen,
+            [out] socklen_t* msg_namelen_out,
+            [out, size=len] void* buf,
+            size_t len,
+            [out, size=msg_controllen] void* msg_control,
+            socklen_t msg_controllen,
+            [out] socklen_t* msg_controllen_out,
+            [out] int* msg_flags,
+            /* -- end struct msghdr -- */
+            int flags)
+            transition_using_threads;
+
+        long myst_shutdown_ocall(int sockfd, int how);
+
+        long myst_listen_ocall(int sockfd, int backlog);
+
+        long myst_getsockname_ocall(
+            int sockfd,
+            [out, size=addr_size] struct sockaddr* addr,
+            [in, out] socklen_t* addrlen,
+            socklen_t addr_size);
+
+        long myst_getpeername_ocall(
+            int sockfd,
+            [out, size=addr_size] struct sockaddr* addr,
+            [in, out] socklen_t* addrlen,
+            socklen_t addr_size);
+
+        long myst_socketpair_ocall(
+            int domain,
+            int type,
+            int protocol,
+            [out] int sv[2]);
+
+        long myst_setsockopt_ocall(
+            int sockfd,
+            int level,
+            int optname,
+            [in, size=optlen] const void* optval,
+            socklen_t optlen);
+
+        long myst_getsockopt_ocall(
+            int sockfd,
+            int level,
+            int optname,
+            [out, size=optval_size] void* optval,
+            [in, out] socklen_t* optlen,
+            socklen_t optval_size);
+
+        // ATTN: If Host file system support is exclude in certain build mode,
+        // consider excluding relevant OCALL proxy code too.
+
+        /*
+        **======================================================================
+        **
+        ** hostfs
+        **
+        **======================================================================
+        */
+
+        long myst_ioctl_ocall(
+            int fd,
+            unsigned long request,
+            [in, out, size=argp_size] void* argp,
+            size_t argp_size);
+
+        long myst_open_ocall(
+            [in, string] const char* pathname,
+            int flags,
+            mode_t mode,
+            uid_t uid,
+            gid_t gid);
+
+        long myst_read_ocall(
```

Comment:
FYI: we can actually do something like
```
#ifdef GCOV
    long myst_read_ocall(
            int fd,
            [out, size=count] void* buf,
            size_t count);
#else
    long myst_read_ocall(
            int fd,
            [out, size=count] void* buf,
            size_t count)
    transition_using_threads;
#endif
```
Just need to do
```
EDGER8R_OPTS += -DGCOV
```
in the makefile to define the variable.